flask_jwt_extended/utils.py

```diff
@@ -82,6 +83,12 @@ def decode_token(encoded_token, csrf_value=None, allow_expired=False):
     unverified_headers = jwt.get_unverified_header(encoded_token)
     # Attempt to call callback with both claims and headers, but fallback to just claims
     # for backwards compatibility
+
+    # Added possibility to disable audience check
+    verify_audience = True
+    if config.audience is None:
+        verify_audience = False
```

Comment:
I don't like this. The current default is having no audiences configured (and thus raising an exception if there is an audience in the token), whereas this changes the current behavior to never check audiences even if there is an audience in the token.

Instead, I think this would be better as an entirely different config option, `JWT_VERIFY_AUDIENCE`.